File: woodroid/src/main/java/me/gilo/woodroid/data/api/OrderNoteAPI.java
package me.gilo.woodroid.data.api;


import me.gilo.woodroid.models.OrderNote;
import retrofit2.Call;
import retrofit2.http.*;

import java.util.List;
import java.util.Map;

public interface OrderNoteAPI {

    @Headers("Content-Type: application/json")
    @POST("orders/{id}/notes")
    Call<OrderNote> create(@Path("id") int order_id, @Body OrderNote body);

    @GET("orders/{id}/notes/{note_id}")
    Call<OrderNote> view(@Path("id") int order_id, @Path("note_id") int note_id);

    @GET("orders/{id}/notes")
    Call<List<OrderNote>> list(@Path("id") int order_id);

    @DELETE("orders/{id}/notes/{note_id}")
    Call<OrderNote> delete(@Path("id") int order_id, @Path("note_id") int note_id);

    @DELETE("orders/{id}/notes/{note_id}")
    Call<OrderNote> delete(@Path("id") int order_id, @Path("note_id") int note_id, @Query("force") boolean force);

    @GET("orders/{id}/notes")
    Call<List<OrderNote>> filter(@Path("id") int order_id, @QueryMap Map<String, String> filter);

}

File: woodroid/src/main/java/me/gilo/woodroid/data/api/OrderNoteAPI.kt
package me.gilo.woodroid.data.api


import me.gilo.woodroid.models.OrderNote
import retrofit2.Call
import retrofit2.http.*

interface OrderNoteAPI {

    @Headers("Content-Type: application/json")
    @POST("orders/{id}/notes")
    fun create(@Path("id") order_id: Int, @Body body: OrderNote): Call<OrderNote>

    @GET("orders/{id}/notes/{note_id}")
    fun view(@Path("id") order_id: Int, @Path("note_id") note_id: Int): Call<OrderNote>

    @GET("orders/{id}/notes")
    fun list(@Path("id") order_id: Int): Call<List<OrderNote>>

    @DELETE("orders/{id}/notes/{note_id}")
    fun delete(@Path("id") order_id: Int, @Path("note_id") note_id: Int): Call<OrderNote>

    @DELETE("orders/{id}/notes/{note_id}")
    fun delete(@Path("id") order_id: Int, @Path("note_id") note_id: Int, @Query("force") force: Boolean): Call<OrderNote>

    @GET("orders/{id}/notes")
    fun filter(@Path("id") order_id: Int, @QueryMap filter: Map<String, String>): Call<List<OrderNote>>

}